akka-typed/src/main/scala/akka/typed/javadsl/ActorContext.scala

```diff
@@ -0,0 +1,160 @@
+ * on an [[ActorRef]], the second is provided by [[ActorContext#spawn]]
+ * and the third is implicit in the signature of [[Behavior]] in that the next
+ * behavior is always returned from the message processing logic.
+ *
+ * An `ActorContext` in addition provides access to the Actor’s own identity (“`getSelf`”),
+ * the [[ActorSystem]] it is part of, methods for querying the list of child Actors it
+ * created, access to [[Terminated DeathWatch]] and timed message scheduling.
+ */
+@DoNotInherit
+@ApiMayChange
+trait ActorContext[T] {
+  // this must be a pure interface, i.e. only abstract methods
+
+  /**
+   * The identity of this Actor, bound to the lifecycle of this Actor instance.
+   * An Actor with the same name that lives before or after this instance will
+   * have a different [[ActorRef]].
+   */
+  def getSelf: ActorRef[T]
+
+  /**
+   * Return the mailbox capacity that was configured by the parent for this actor.
+   */
+  def getMailboxCapacity: Int
+
+  /**
+   * The [[ActorSystem]] to which this Actor belongs.
+   */
+  def getSystem: ActorSystem[Void]
+
+  /**
+   * The list of child Actors created by this Actor during its lifetime that
+   * are still alive, in no particular order.
+   */
+  def getChildren: java.util.List[ActorRef[Void]]
+
+  /**
+   * The named child Actor if it is alive.
+   */
+  def getChild(name: String): Optional[ActorRef[Void]]
+
+  /**
+   * Create a child Actor from the given [[akka.typed.Behavior]] under a randomly chosen name.
+   * It is good practice to name Actors wherever practical.
+   */
+  def spawnAnonymous[U](behavior: Behavior[U]): ActorRef[U]
+
+  /**
+   * Create a child Actor from the given [[akka.typed.Behavior]] under a randomly chosen name.
+   * It is good practice to name Actors wherever practical.
+   */
+  def spawnAnonymous[U](behavior: Behavior[U], deployment: DeploymentConfig): ActorRef[U]
+
+  /**
+   * Create a child Actor from the given [[akka.typed.Behavior]] and with the given name.
+   */
+  def spawn[U](behavior: Behavior[U], name: String): ActorRef[U]
+
+  /**
+   * Create a child Actor from the given [[akka.typed.Behavior]] and with the given name.
+   */
+  def spawn[U](behavior: Behavior[U], name: String, deployment: DeploymentConfig): ActorRef[U]
+
+  /**
+   * Force the child Actor under the given name to terminate after it finishes
+   * processing its current message. Nothing happens if the ActorRef does not
+   * refer to a current child actor.
+   *
+   * @return whether the passed-in [[ActorRef]] points to a current child Actor
+   */
+  def stop[U](child: ActorRef[U]): Boolean
+
+  /**
+   * Register for [[Terminated]] notification once the Actor identified by the
+   * given [[ActorRef]] terminates. This notification is also generated when the
+   * [[ActorSystem]] to which the referenced Actor belongs is declared as
+   * failed (e.g. in reaction to being unreachable).
+   */
+  def watch[U](other: ActorRef[U]): Unit
```

Comment:
I'll get to @raboof's PR in a bit, but asking right away - where is the watchWith going to be defined?